ir/memory.cpp

```diff
@@ -58,17 +58,36 @@ static bool is_constglb(unsigned bid) {
   return false;
 }
 
+// Return true if bid is the nonlocal block used to encode function calls' side
+// effects
+static bool is_fncall_mem(unsigned bid) {
+  if (!has_fncall)
+    return false;
+  return bid == num_globals_src + num_ptrinputs + 1;
+}
+static unsigned get_fncallmem_bid() {
+  assert(has_fncall);
+  return num_globals_src + num_ptrinputs + 1;
+}
+void ensure_non_fncallmem(const Pointer &p) {
+  if (!p.isLocal().isFalse())
+    return;
+  uint64_t ubid;
+  ENSURE(!p.getShortBid().isUInt(ubid) || !is_fncall_mem(ubid));
```

Comment:
should be an assert. ENSURE is assert + guaranteed side-effects

Reply:
ENSURE is used here to resolve unused variable warning of ubid.

Reply:
but that adds overhead to non-debug builds. just add a (void) after ubid f.ex.